### PR TITLE
Support receiving own proposals without an error

### PR DESCRIPTION
**Pull request by mulmarta**
To make implementing DS's easier, this PR allows clients to receive their own proposals.
- For plaintext proposals, this is trivial, since they can be processed the same way as proposals from other clients
- For ciphertext proposals, we cache a list of hashes of all encrypted proposals we generated and just return Ok when processing one of them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT license.

### CODE REVIEW

File: mls-rs/src/group/proposal_cache.rs [hunk 1]
```diff
@@ -169,6 +203,17 @@ impl ProposalCache {
...
+
+    #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+    pub async fn contains_own<CS: CipherSuiteProvider>(
+        &self,
+        cs: &CS,
+        message: &MlsMessage,
+    ) -> Result<bool, MlsError> {
+        let message_hash = MessageHash::compute(cs, message).await?;
+
+        Ok(self.own_proposals.iter().any(|op| op == &message_hash))
```
Comment (stefunctional) [inline mls-rs/src/group/proposal_cache.rs:215]: Nit:
```suggestion
        Ok(self.own_proposals.contains(&message_hash))
```

File: mls-rs/src/group/proposal_cache.rs [hunk 2]
```diff
@@ -46,14 +51,24 @@ pub struct CachedProposal {
...
-#[derive(Clone, PartialEq)]
+#[derive(Clone)]
 pub(crate) struct ProposalCache {
     protocol_version: ProtocolVersion,
     group_id: Vec<u8>,
     #[cfg(feature = "std")]
     pub(crate) proposals: HashMap<ProposalRef, CachedProposal>,
     #[cfg(not(feature = "std"))]
     pub(crate) proposals: Vec<(ProposalRef, CachedProposal)>,
+    pub(crate) own_proposals: Vec<MessageHash>,
```
Comment (stefunctional) [inline mls-rs/src/group/proposal_cache.rs:62]: Does this ever get cleared?

Comment (mulmarta) [reply mls-rs/src/group/proposal_cache.rs:62]: 🤦 Thank you!

**Review by stefunctional — APPROVED** [review]
Thank you!

File: mls-rs/src/group/proposal_cache.rs (hunk 2 above)
Comment (stefunctional) [reply mls-rs/src/group/proposal_cache.rs:62]: You're welcome!

**Review by stefunctional — DISMISSED** [review]

**Review by stefunctional — DISMISSED** [review]

**Review by stefunctional — APPROVED** [review]

**Review by tomleavy — APPROVED** [review]

**Pull request merged**